app/src/command.rs

```diff
@@ -285,19 +301,85 @@ impl<'app, D: Dependencies<'app>> Application<'app, D> {
         histogram!("block_build_duration_seconds").record(t0.elapsed().as_secs_f64());
         metrics::counter!("blocks_produced_count").increment(1);
 
-        Ok(())
+        Ok(Some(block_hash))
     }
 
     pub fn add_transaction(&mut self, tx: NormalizedEthTransaction) {
         self.mem_pool.insert(tx);
     }
 
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub(crate) fn forkchoice_update(
```

Comment:
Nit: we could have a docstring here too, if there's a new one on block build already.